proposed/swift-interop.md

```diff
@@ -152,33 +152,46 @@ If possible, Swift tuples should be represented as `ValueTuple`s in .NET. If thi
+
+The projection tooling should be split into these components.
 
 ##### Importing Swift into .NET
 
-1. A tool that takes in a `.swiftinterface` file or Swift sources and produces C# code.
-2. A library that provides the basic support for Swift interop that the generated code builds on.
-3. User tooling to easily generate Swift projections for a given set of `.framework`s.
-    - This tooling would build a higher-level interface on top of the tool in item 1 that is more user-friendly and project-system-integrated.
-4. (optional) A NuGet package, possibly referencable by `FrameworkReference` or automatically included when targeting macOS, Mac Catalyst, iOS, or tvOS platforms that exposes the platform APIs for each `.framework` that is exposed from Swift to .NET.
-    - This would be required to provide a single source of truth for Swift types so they can be exposed across an assembly boundary.
+Importing Swift into .NET is done through bindings generated by the following components:
+ - `Demangler`: This component analyzes a source Swift library and maps entry points to mangled names. It consumes a source Swift library and creates `TLDefinition` objects that contain type, mangled name, demangled name, module name, and offset within a module.
+ - `Swiftmodule parser`: This component aggregates the public API. It takes in a `.swiftmodule` file generated by the Swift compiler, similar to a C header file, and combines it with the `TLDefinition` objects to generate a module declaration.
+ - `WrappingCompiler` and `OverrideBuilder`: These components generate Swift wrappers for cases where direct P/Invoke from C# into Swift is not possible.
+ - `NewClassCompiler`: This component generates C# bindings based on the generated `TLDefinition` and module declarations. It should implement type mapping and register lowering. Key considerations include:
+   - Splitting out tuples into individual parameters
+   - Determining the return buffer size using the value witness table for the struct type
+   - Managing enum/struct arguments through registers vs. passing by reference
+   - Error handling
+ - `XamGlue`: This component is a library that provides basic support for Swift interop that the generated code builds on. It includes bindings for common built-in types like arrays and dictionaries. It provides a single source of truth for Swift types so that they can be exposed across an assembly boundary. This can be a NuGet package, possibly referencable by `FrameworkReference` or automatically included when targeting macOS, Mac Catalyst, iOS, or tvOS platforms that exposes the platform APIs for each `.framework` that is exposed from Swift to .NET.
+ - `Tom-Swifty`: This component orchestrates other components and allows users to easily generate Swift projections for a given set of `.framework`s. It presents an interface on top of the tool that is user-friendly and project-system-integrated.
```

Comment:
What does Tom stand for?

Reply:
It is just a name of the component, perhaps a wordplay? :)